picodom.d.ts

```diff
@@ -1,23 +1,38 @@
-export interface VNode<Data> {
+export as namespace picodom
+
+export interface VNode<Props> {
   type: string
   props?: Props
-  children: VNode<Props> | string
+  children: Array<VNode<{}> | string>
 }
 
-export interface Component<Props> {
-  (props?: Props, ...children: VNode<Props> | string): VNode<Props>
+export interface Component<Props = {}> {
+  (props: Props, children: Array<VNode<{}> | string>): VNode<Props>
 }
 
 export function h<Props>(
   type: Component<Props> | string,
   props?: Props,
-  ...children: VNode<Props> | string
+  ...children: Array<VNode<{}> | string | number>
+): VNode<Props>
+
+export function h<Props>(
+  tag: Component<Props> | string,
+  data?: Props,
+  children?: Array<VNode<{}> | string | number>
 ): VNode<Props>
 
 export function patch(
   oldNode: VNode<{}> | null,
   newNode: VNode<{}>,
-  container: HTMLElement = document.body
+  container?: HTMLElement
 ): Element
 
-export as namespace picodom
+declare global {
+  namespace JSX {
+    interface Element extends VNode<any> {}
```

Comment:
Note: I explicitly don't make Element generic as doing that causes TypeScript to no longer enforce JSX prop types.